pkg/agent/flowexporter/connections/connections.go

```diff
@@ -240,71 +66,37 @@ func (cs *connectionStore) ForAllConnectionsDo(callback flowexporter.ConnectionM
 	return nil
 }
 
-// Poll calls into conntrackDumper interface to dump conntrack flows. It returns the number of connections for each
-// address family, as a slice. In dual-stack clusters, the slice will contain 2 values (number of IPv4 connections first,
-// then number of IPv6 connections).
-// TODO: As optimization, only poll invalid/closed connections during every poll, and poll the established connections right before the export.
-func (cs *connectionStore) Poll() ([]int, error) {
-	klog.V(2).Infof("Polling conntrack")
-	// Reset IsPresent flag for all connections in connection map before dumping flows in conntrack module.
-	// if the connection does not exist in conntrack table and has been exported, we will delete it from connection map.
-	deleteIfStaleOrResetConn := func(key flowexporter.ConnectionKey, conn *flowexporter.Connection) error {
-		if !conn.IsPresent && conn.DoneExport {
-			if err := cs.deleteConnectionByKeyWithoutLock(key); err != nil {
-				return err
-			}
-		} else {
-			conn.IsPresent = false
-		}
-		return nil
+func (cs *connectionStore) fillPodInfo(conn *flowexporter.Connection) {
+	if cs.ifaceStore == nil {
+		klog.Warning("Interface store is not available to retrieve local Pods information.")
+		return
 	}
```

Comment:
out-of curiosity, is there an error scenario in which this can happen? Hopefully not, as I imagine the logs would be flooded with this warning

Reply:
I added this for easier testing purpose. It should not happen. Changed the logging verbose just in case.